keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -459,87 +463,123 @@ Windows privileged containers will be implemented with [Job Objects](https://doc
+  string credential_spec = 2;
   bool privileged = 3;
 }
 ```
-### Kubelet Implementation Details
 
-#### <b>Privileged Flag</b>
-There are no kubernetes API changes required.  Kubelet can use the privileged flag from SecurityContext api and pass that to the new privileged flag in the CRI layer.  
+*Note:* For alpha annotaions on RunPodSandbox and CreateContainer CRI calls may be used untill a version of containerd with Windows privileged container support is released.
+
+### Kubernetes API updates
+
+#### Privilged Flag
+
+A new boolean field named `privileged` will be added to [WindowsSecurityContextOptions](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#windowssecuritycontextoptions-v1-core).
+
+On Windows, all containers in a pod mush be privileged. Because of this behavior and because `WindowsSecurityContextOptions` already exists on both `PodSecurityContext` and `Container.SecurityContext` Windows containers will use this new field instead of re-using the existing `privileged` field which only exists on `SecurityContext`.
 
-Add functionality to Kuberuntime_sandbox to split out the linux sandbox creation and add windows sandbox creation (WIP PR). 
+Having a new field will also help us better anticipate / validate user intention.
 
-https://github.com/kubernetes/kubernetes/blob/a9f1d72e1de6450b890a0c0e451725468f54f515/pkg/kubelet/kuberuntime/kuberuntime_sandbox.go#L136 
+API validation will ensure that if `privileged` is set in `Container.SecurityContext.WindowsSercurityContextOptions` then `privileged` is also set in `PodSecurityContext.WindowsSecurityContextOptions`. Current behavior applies pod `WindowsSecurityContextOptions` to all container level `WindowsSecurityContextOptions` if not specified.
 
-#### <b>Host Network Mode</b>
+No additional validation will be performed against the existing `privileged` on `SecurityContext`.
 
-If using Windows privileged containers the Host Network Mode will also be enabled, as the pod will automatically get the host IP.  The PodSpec has a hostNetwork boolean field:
+#### Host Network Mode
 
-https://github.com/kubernetes/kubernetes/blob/d20fd4088476ec39c5ae2151b8fffaf0f4834418/pkg/kubelet/container/helpers.go#L227-L229
+Host Network mode for privleged Windows containers will always be enabled, as the pod will automatically get the host IP.
 
-https://github.com/kubernetes/kubernetes/blob/a9f1d72e1de6450b890a0c0e451725468f54f515/pkg/kubelet/kuberuntime/kuberuntime_sandbox.go#L98 
+Privileged Windows containers will be unable to align to pod namespaces due to limitiations in the Windows OS. This functionaility may likely be enabled in the future through a new KEP.
+
+Because of this we do not plan on performing **any** valiation around the existing `hostNetowrk` PodSpec field. This will give us the flexability to align privileged containers to host or pod namespaces in the future if/when that functionaility is added. Instead we will rely on documentation to explain 
```

Comment:
why not require spec.hostNetwork=true for privileged windows containers until that functionality is added? that would keep the API content accurately reflecting the nature of the pod

Reply:
I had two thoughts here.

First - In the future we hope to support scenarios where Windows privileged containers could be joined to pod networks. 
If it would be ok to loosen validation (allow privileged=true spec.hostNetwork=false) then I would be OK with ensuring spec.hostNetwork = true when for privileged containers.

Second - I am a little worried that using privileged containers on Windows will require setting many fields in the specs and that this would become cumbersome.

Open to changing this to perform more validation tho :)

Reply:
Just realized this contradicts the statement below.
I'll update the enhancement to say we'll enforce hostNetwork = true if privileged = true until future updates come and we can support privileged containers being joined to a pod network.